gobblin-core/src/main/java/gobblin/source/extractor/extract/kafka/KafkaAvroExtractor.java

```diff
@@ -0,0 +1,140 @@
+/* (c) 2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.source.extractor.extract.kafka;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+
+import kafka.message.MessageAndOffset;
+
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericData.Record;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.io.Decoder;
+import org.apache.avro.io.DecoderFactory;
+
+import com.google.common.base.Preconditions;
+
+import gobblin.configuration.WorkUnitState;
+import gobblin.source.extractor.DataRecordException;
+
+
+public class KafkaAvroExtractor extends KafkaExtractor<Schema, GenericRecord> {
+
+  public static final String KAFKA_SCHEMA_REGISTRY_URL = "kafka.schema.registry.url";
+  public static final int SCHEMA_ID_LENGTH_BYTE = 16;
+  private static final byte MAGIC_BYTE = 0x0;
+
+  private final Schema schema;
+  private final KafkaAvroSchemaRegistry schemaRegistry;
+
+  public KafkaAvroExtractor(WorkUnitState state) {
+    super(state);
+    Preconditions.checkArgument(state.contains(KAFKA_SCHEMA_REGISTRY_URL), "Schema registry URL not provided.");
+    this.schemaRegistry = new KafkaAvroSchemaRegistry(state.getProp(KAFKA_SCHEMA_REGISTRY_URL));
+    this.schema = getLatestSchemaByTopic();
+  }
+
+  private Schema getLatestSchemaByTopic() {
+    try {
+      return this.schemaRegistry.getLatestSchemaByTopic(this.partition.getTopicName());
+    } catch (SchemaNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public Schema getSchema() {
+    return this.schema;
+  }
+
+  @Override
+  public GenericRecord readRecord(GenericRecord reuse) throws DataRecordException, IOException {
```

Comment:
Seems this method is exactly the same as the one in `KafkaExtractor`, except for the return type that is concrete. You don't need to override. The one in `KafkaExtractor` will return `GenericRecord` when called with `KafkaAvroExtractor `.